.github/workflows/ci.yml

```diff
@@ -57,11 +52,62 @@ jobs:
       - name: Build test place
         run: rojo build test-runner.project.json -o test.rbxl
 
+      - name: Download OpenVPN
+        run: choco install openvpn
+
+      - name: Run OpenVPN
+        run: Start-Process -FilePath "C:\\Program Files\\OpenVPN\\bin\\openvpn.exe" -ArgumentList "--config $((Resolve-Path .\\fusion.ovpn).Path)"
+
+      - name: Poll for IP Change
+        run: |
+          $elapsed = 0
+          while ($true) {
+            try {
+              $response = Invoke-WebRequest -Uri 'https://httpbin.org/ip' -Method GET -UseBasicParsing
+              $content = ConvertFrom-Json $response.Content
+              if ($content.origin -eq "104.238.130.74") {
+                break
+              }
+            } catch {}
+            if ($elapsed -ge 25) {
+              Write-Error "Timeout reached!"
+              exit 1
+            }
+            Write-Output "Polling.. Elasped: $elapsed, IP: $($content.origin)"
+            Start-Sleep 5
+            $elapsed += 5
+          }
+          Write-Output "Success!"
+
+      - name: Validate Cookie
+        run: |
+          $session = New-Object Microsoft.PowerShell.Commands.WebRequestSession
+          $cookie = New-Object System.Net.Cookie
+          $cookie.Name = ".ROBLOSECURITY"
+          $cookie.Value = "${{ env.ROBLOSECURITY }}"
+          $cookie.Domain = ".roblox.com"
+          $session.Cookies.Add($cookie);
+          Invoke-WebRequest "https://avatar.roblox.com/v1/avatar" -WebSession $session -UseBasicParsing
+
+      - name: Install Roblox Studio
+        uses: OrbitalOwen/roblox-win-installer-action@1.1
+        with:
+          # This cookie is intentionally public, because secrets are not available in pull request CI runs
+          # It is from an empty account and not a security vulnerability
```

Comment:
This comment doesn't make much sense here anymore. Could it be moved to the new place for the cookie?